openquake/hazardlib/gsim/panckow_pechmann_2004.py

```diff
@@ -0,0 +1,161 @@
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`PanckowPechmann2004`.
+"""
+from __future__ import division
+
+import numpy as np
+from scipy.constants import g
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV
+
+class PanckowPechmann2004(GMPE):
+    """
+    Description
+    """
+    #: Supported tectonic region type is active shallow crust,
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: TO CHECK PSV!
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV
+    ])
+
+    #: Supported intensity measure component is VECTORIAL
+    #: :attr:`~openquake.hazardlib.const.IMC.VECTORIAL`,
+    #: NOTE: The paper indicates it as Geometric mean (to check)
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.VECTORIAL
+
+    #: Supported standard deviation type is total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL
+    ])
+
+    #: Required site parameter is only Vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30', ))
+
+    #: Required rupture parameter is magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', ))
+
+    #: Required distance measure is Rjb distance
+    #: see paragraph 'Predictor Variables', page 6.
+    REQUIRES_DISTANCES = set(('rjb', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        C = self.COEFFS[imt]
+
+        mag = rup.mag - 6
+        d = np.sqrt(dists.rjb ** 2 + C['h'] ** 2)
+
+        # In the original formulation of the GMPE, distinction is only made
+        # between rock and soil sites, which I assumed are separated by the Vs30
+        # value of 910m/s (see equation 5 of the paper)
+        gamma = 0 if sites.vs30.any() > 910 else 1
+
+        mean = np.zeros_like(d)
+
+        mean += C['b1'] + \
+                C['b2'] * mag + \
+                C['b3'] * mag ** 2 + \
+                C['b5'] * np.log10(d) + \
+                C['b6'] * gamma
+
+        # Convert from base 10 to base e
+        mean = np.log(10.0 ** mean)
+
+        # idx = d > 100.
+        # mean[idx] = 0
+
+        stddevs = self._get_stddevs(C, stddev_types,  dists.rjb.shape[0])
```

Comment:
It looks like you are returning the stddevs in terms of log_10 rather than log_e. This needs to be adjusted too.